### ISSUE TITLE
First launch segfault

### ISSUE THREAD

**Issue by lshprung** [lshprung/tml#2]
Current testing branch can be built for windows, but crashes somewhere between finding the config file and initscr.

**Comment by lshprung** [comment]
This does not appear to occur when debug symbols are turned on. Debugging will be challenging, so I will put this off for now

**Comment by lshprung** [comment]
Update: this occurs on Linux as well. It does not occur if built with debug symbols (`CFLAGS=-g`). Some warnings may help to debug:

```
read_cfg.c: In function ‘autoAlias’:
read_cfg.c:508:31: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  508 |  else if(*path == '"') *(p-1) = '\0'; //close early to avoid including closing quote
      |                        ~~~~~~~^~~~~~
read_cfg.c:458:18: note: at offset -1 to an object with size 0 allocated by ‘malloc’ here
  458 |  char *hr_name = malloc(sizeof(char) * BUF_LEN);
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
unix/read_cfg.c: In function ‘find_config’:
cc1: warning: function may return address of local variable [-Wreturn-local-addr]
unix/read_cfg.c:19:7: note: declared here
   19 |  char choices[2][BUF_LEN];
      |       ^~~~~~~
cc1: warning: function may return address of local variable [-Wreturn-local-addr]
unix/read_cfg.c:19:7: note: declared here
unix/read_cfg.c:19:7: note: declared here
```

**Comment by lshprung** [comment]
Fixed in 01d60765f8441e5d81e7c02770731af8c0676c21 and 1f29cdb08e29f86053a813ff61b617f5994f938b
